lib/doorkeeper/oauth/client_credentials/creator.rb

```diff
@@ -11,7 +11,9 @@ def call(client, scopes, attributes = {})
             return existing_token
           end
 
-          existing_token&.revoke
+          if Doorkeeper.configuration.revoke_previous_client_credentials_token
```

Comment:
Could we add a new private method called `configuration` as a synthetic sugar for accessing `Doorkeeper.configuration` ? Maybe a delegator I don't know. I see we call it multiple times for this module, we can simplify some constructions.

Reply:
done